doc/api/cluster.markdown

```diff
@@ -2,16 +2,44 @@
 
     Stability: 2 - Stable
 
-A single instance of Node.js runs in a single thread. To take advantage of
-multi-core systems the user will sometimes want to launch a cluster of Node.js
-processes to handle the load.
+Node.js executes JavaScript code within the context of a single-threaded event
```

Comment:
I took a stab at making the cluster introduction more approachable to beginners. I was able to consolidate three paragraphs down to two which is a win with respects to the reader seeing the first code example. In addition, I tried to clearly answer why clusters are important by first explaining the benefits and shortcomings of `libuv` at a high level.

In the process, the references about `child_process.fork()` and IPC channels was cut. In my experience as a teacher, a newcomer wants to know three things in the beginning—what a thing is, why it's important, and how to use it. Implementation details can come (much) later.

Anyway, feel free to use none, some, or all of it.  :)

---

Node.js executes JavaScript code within the context of an event loop. Under the hood, non-blocking socket I/O runs on the main event loop thread. On the other hand, blocking DNS and filesystem I/O runs on secondary threads that notify the main thread when finished. This currency model allows an [I/O bound](https://en.wikipedia.org/wiki/I/O_bound) application to easily distribute work across multiple CPU cores. For more details, see the [libuv](http://docs.libuv.org/en/v1.x/design.html) project.

Additionally, the `cluster` API allows a [CPU bound](https://en.wikipedia.org/wiki/CPU-bound) application to distribute its work across multiple CPU cores. A Node.js cluster is created when a **master process** forks one or more **worker processes**. By design, workers use the same application code as the master. Conveniently, the `cluster.isMaster` property determines if a running process is the master or a worker.

    const cluster = require('cluster');
    if (cluster.isMaster) {
      for (var i = 0; i < numCPUs; i++) {
        cluster.fork();
      }
    } else {
      // Distributed work
    }